### PR TITLE
Make BrowserLock start/stop no need to restart

### PR DESCRIPTION
**Pull request by knj0426**
When enable/disable BrowserLock, if user cancel restart application,
BrowserLock is not applied properly because TripleBananaApplication is
not called when next restart.

Signed-off-by: Nojeong Gwak <knj0426@gmail.com>

### CODE REVIEW

**Review by romandev — COMMENTED** [review]
Thank you for this patch.

Your patch looks good to me overall but it has two commits.
If possible, we prefer to put one commit into one PR.
Could you please separate the dark mode related commit from this PR?

**Comment by romandev** [comment]
Also, we don't use "Signed-off" in commit description.

File: src/org/triple/banana/settings/ExtensionFeatures.java [hunk 1]
```diff
@@ -75,6 +75,7 @@ public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
                 (SwitchPreferenceCompat) findPreference(FeatureName.DARK_MODE);
         darkMode.setChecked(DarkModeController.get().isDarkModeOn());
         darkMode.setOnPreferenceChangeListener((preference, newValue) -> {
+            BrowserLock.setExceptional(true);
```
Comment (jdragonbae) [inline src/org/triple/banana/settings/ExtensionFeatures.java:78]: Does this mean impossible on dark mode temporary?

Comment (knj0426) [reply src/org/triple/banana/settings/ExtensionFeatures.java:78]: Yes. When toggle dark mode, application is restarted forcely. So, ApplicationStatus is changed background and foreground in a very short time. ApplicationApplicationTracker cannot avoid this behavior, so I though it should be handled as exceptional case.

**Comment by knj0426** [comment]
> Thank you for this patch.
> 
> Your patch looks good to me overall but it has two commits.
> If possible, we prefer to put one commit into one PR.
> Could you please separate the dark mode related commit from this PR?

**Comment by knj0426** [comment]
> Thank you for this patch.
> 
> Your patch looks good to me overall but it has two commits.
> If possible, we prefer to put one commit into one PR.
> Could you please separate the dark mode related commit from this PR?

Ok. I will do it

**Pull request closed without merging**